geth/api/app_state.go

```diff
@@ -0,0 +1,39 @@
+package api
+
+import (
+	"fmt"
+	"strings"
+)
+
+// AppState represents if an app is in foreground or not,
+// see https://facebook.github.io/react-native/docs/appstate.html
+type AppState string
+
+func (a AppState) String() string {
+	return string(a)
+}
+
+const (
+	AppStateForeground = AppState("foreground")
+	AppStateBackground = AppState("background")
+	AppStateInactive   = AppState("inactive")
+
+	AppStateInvalid = AppState("")
+)
+
+func validAppStates() []AppState {
+	return []AppState{AppStateInactive, AppStateBackground, AppStateForeground}
+}
+
+func ParseAppState(stateString string) (AppState, error) {
```

Comment:
Call me pedantic, but at least for all public functions, types, etc. I would like to see comments. Simply imagine the impression of people reading about our code at https://godoc.org/.

Reply:
Hmm, I thought I fixed that. Maybe outdated diff?